features/weak-references.yml

```
@@ -0,0 +1,14 @@
+name: Weak references
+description: The `WeakRef` and `FinalizationRegistry` objects manage weakly held objects and define cleanup operations for garbage-collected objects.
```

Ugh this is a hard thing to describe. I took another try at this after scratching my head and reading the docs:

```suggestion
description: The `WeakRef` and `FinalizationRegistry` objects hold references to garbage-collectable objects without creating strong references that prevent their garbage collection.
```

That said, I'm not sure this is quite complete. I was thinking maybe adding examples might illustrate the idea better. Something like "For example, these objects can be used to build caches or reference DOM nodes that do not leak memory."

---

Thanks Daniel, this reads good to me. I like the "For example, " sentence as well. Not sure we generally want these in descriptions, though? I got tempted in other PRs, so not sure if we have a guideline or not.

---

We don't have a guideline, but I'm not sure I'd have a good one to offer. We use "For example" in a lot of different ways:

- To make a very abstract or obscure feature more legible (I'm thinking that this is the case here)
- To explain syntax without resorting to awkward metasyntactic variables (e.g., the `custom-properties` CSS feature)
- To describe a feature's options where an exhaustive list would be too long or overly specific (e.g., `font-variant-numeric`, `show-picker-input`). We also do this with "such as."
- To make a circular description a little less circular (e.g., `read-write-pseudos`)
- To explain open-ended behavior that's expected to vary significantly between browsers (e.g., `writingsuggestions`)

Feels like a judgment call to me. 🤷‍♂️

---

Yeah, I think we can leave this as is for now. I would imagine people to go elsewhere (MDN) to find more meaningful information. If there is a web-features consumer story about being more illustrative, we can probably add more words.

---

Sounds good. Thank you!

---

Now that you say it, this particular feature is a better argument for a docs URL field rather than more description text!